react/ProductKit.js

```diff
@@ -1,190 +1,57 @@
-          default: false,
-          isLayout: true,
-        },
-        showBadge: {
-          type: 'boolean',
-          title: 'editor.productKit.showBadge',
-          default: false,
-          isLayout: true,
-        },
-        badgeText: showBadge
-          ? {
-              type: 'string',
-              title: 'editor.productKit.badgeText',
-              isLayout: false,
-            }
-          : {},
-      },
-    }
-  }
-
-  /**
-   * Extract and format the required information of a Product to be used into the
-   * ProductKitItem component.
-   */
-  prepareProductKit = productKit => {
-    const { benefitProduct, discount, minQuantity } = productKit
-    const newProduct = { ...benefitProduct }
+export default class ProductKit extends Component {
+  static propTypes = ProductKitPropTypes.props
 
-    if (newProduct.items && newProduct.items.length) {
-      newProduct.sku = { ...newProduct.items[0] }
-      newProduct.productName = newProduct.sku.nameComplete
-      if (newProduct.sku.sellers && newProduct.sku.sellers.length) {
-        newProduct.sku.seller = newProduct.sku.sellers[0]
-      } else {
-        newProduct.sku.seller = {
-          commertialOffer: {
-            Price: 0,
-            ListPrice: 0,
-          },
-        }
-      }
-      if (newProduct.sku.images && newProduct.sku.images.length) {
-        newProduct.sku.image = { ...newProduct.sku.images[0] }
-        newProduct.sku.image.imageUrl = newProduct.sku.image.imageUrl.replace(
-          /^https?:/,
-          ''
-        )
-      }
-      newProduct.sku.referenceId = (newProduct.sku.referenceId &&
-        newProduct.sku.referenceId[0]) || {
-        Value: '',
-      }
-      delete newProduct.sku.sellers
-      delete newProduct.sku.images
-      delete newProduct.items
-    }
-
-    return { ...newProduct, discount, minQuantity }
-  }
+  static defaultProps = ProductKitPropTypes.defaultProps
 
-  prepareProductSKUKit = productKit => {
-    let productSKUKit = []
-    productKit.forEach(productKitItem => {
-      const { benefitProduct: product, benefitSKUIds } = productKitItem
-      benefitSKUIds.forEach(skuId => {
-        product.items.forEach(item => {
-          if (skuId === item.itemId) {
-            productSKUKit.push({
-              ...productKitItem,
-              benefitProduct: { ...product, items: [item] },
-            })
-          }
-        })
-      })
-    })
-    return productSKUKit
-  }
+  static getSchema = ProductKitSchema
 
   render() {
     const {
-      productQuery: { product },
-      showListPrice,
-      showLabels,
-      showInstallments,
+      slug,
       showBadge,
       badgeText,
+      showLabels,
+      showListPrice,
+      showInstallments,
+      productQuery: { product },
     } = this.props
 
     const benefits = path(['benefits'], product)
 
-    if (benefits && benefits.length === 0) {
+    // The product does not have any Kit associated with it, in this
```

Comment:
/** If the product doesn't have any Kit associated with it
then it shouldn't be rendered */